### PR TITLE
Branch find tasks

### PR DESCRIPTION
**Pull request by Kaya3842**
Users can find tasks based on matching `TaskName`. Command format:

`findtask KEYWORD [MORE_KEYWORDS]`

- The search is case-insensitive. e.g 2103t will match 2103T
- The order of the keywords does not matter. e.g. 2101 2103t will match 2103t 2101
- Only the `TaskName` is searched.
- Only full words will be matched e.g. 2103 will not match 2103t
- Tasks matching at least one keyword will be returned (i.e. OR search). e.g. 2101 2103t will return 2101 CA2, 2103t new feature

### CODE REVIEW

**Review by Yskie — APPROVED** [review]
LGTM!

File: src/main/java/seedu/address/logic/commands/MarkTaskCommand.java [hunk 1]
```diff
@@ -47,7 +46,6 @@ public CommandResult execute(Model model) throws CommandException {
         editedTask.getStatus().setAsDone();
 
         model.setTask(taskToMark, editedTask);
-        model.updateFilteredTaskList(PREDICATE_SHOW_ALL_TASKS);
```
Comment (breezetall) [inline src/main/java/seedu/address/logic/commands/MarkTaskCommand.java:50]: Would it be better to display the full list after modifying a task?

File: src/main/java/seedu/address/logic/commands/UnmarkTaskCommand.java [hunk 2]
```diff
@@ -47,8 +46,6 @@ public CommandResult execute(Model model) throws CommandException {
 
         model.setTask(taskToMark, editedTask);
 
-        model.updateFilteredTaskList(PREDICATE_SHOW_ALL_TASKS);
```
Comment (breezetall) [inline src/main/java/seedu/address/logic/commands/UnmarkTaskCommand.java:50]: Same comment as mark task.

**Review by breezetall — APPROVED** [review]
Looks good to me! Just some minor suggestions.

File: src/main/java/seedu/address/model/task/TaskNameContainsKeywordsPredicate.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.StringUtil;
+import seedu.address.commons.util.ToStringBuilder;
+
+/**
+ * Tests that a {@code Task}'s {@code TaskName} matches any of the keywords given.
+ */
+public class TaskNameContainsKeywordsPredicate implements Predicate<Task> {
```
Comment (nobodyishappy) [inline src/main/java/seedu/address/model/task/TaskNameContainsKeywordsPredicate.java:12]: Great implementation of the predicate for checking!

File: src/test/java/seedu/address/logic/commands/FindTaskCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+        // null -> returns false
+        assertFalse(findFirstCommand.equals(null));
+
+        // different person -> returns false
+        assertFalse(findFirstCommand.equals(findSecondCommand));
+    }
+
+    @Test
+    void testToString() {
```
Comment (nobodyishappy) [inline src/test/java/seedu/address/logic/commands/FindTaskCommandTest.java:87]: Should there be a public for this test function?

File: src/test/java/seedu/address/model/task/TaskNameContainsKeywordsPredicateTest.java [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.testutil.TaskBuilder;
+
+class TaskNameContainsKeywordsPredicateTest {
+    @Test
+    void test_taskNameContainsKeywords_returnsTrue() {
```
Comment (nobodyishappy) [inline src/test/java/seedu/address/model/task/TaskNameContainsKeywordsPredicateTest.java:17]: Should there be a public for this test function?

File: src/test/java/seedu/address/model/task/TaskNameContainsKeywordsPredicateTest.java [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+        assertFalse(predicate.test(new TaskBuilder().withTaskName("2103t").build()));
+
+        // Keywords match description, but does not match name
+        predicate = new TaskNameContainsKeywordsPredicate(Arrays.asList("dummy"));
+        assertFalse(predicate.test(new TaskBuilder().withTaskName("2103t")
+                .withTaskDescription("dummy").build()));
+    }
+
+    @Test
+    void testEquals() {
```
Comment (nobodyishappy) [inline src/test/java/seedu/address/model/task/TaskNameContainsKeywordsPredicateTest.java:53]: Should there be a public for this test function?

File: src/test/java/seedu/address/model/task/TaskNameContainsKeywordsPredicateTest.java [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+        // null -> returns false
+        assertFalse(firstPredicate.equals(null));
+
+        // different values -> returns false
+        assertFalse(firstPredicate.equals(secondPredicate));
+    }
+
+    @Test
+    void testToString() {
```
Comment (nobodyishappy) [inline src/test/java/seedu/address/model/task/TaskNameContainsKeywordsPredicateTest.java:81]: Should there be a public for this test function?

File: src/main/java/seedu/address/logic/commands/FindTaskCommand.java [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+ */
+public class FindTaskCommand extends Command {
+    public static final String COMMAND_WORD = "findtask";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all tasks whose names contain any of "
+            + "the specified keywords (case-insensitive) and displays them as a list with index numbers.\n"
+            + "Parameters: KEYWORD [MORE_KEYWORDS]...\n"
+            + "Example: " + COMMAND_WORD + " 2101 2103t";
+
+    private final TaskNameContainsKeywordsPredicate predicate;
```
Comment (nobodyishappy) [inline src/main/java/seedu/address/logic/commands/FindTaskCommand.java:23]: Should this be in all capital letter as it is a final value?

**Review by nobodyishappy — APPROVED** [review]
Other than some minor standardization issues, looks good to me!

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java [hunk 9]
```diff
@@ -82,6 +83,9 @@ public Command parseCommand(String userInput) throws ParseException {
         case FindCommand.COMMAND_WORD:
             return new FindCommandParser().parse(arguments);
 
+        case FindTaskCommand.COMMAND_WORD:
+            return new FindTaskCommandParser().parse(arguments);
```
Comment (chin-herng) [inline src/main/java/seedu/address/logic/parser/AddressBookParser.java:87]: Might be better to add a test in `AddressBookParserTest` to ensure that the new command is being parsed correctly!

**Review by chin-herng — APPROVED** [review]
LGTM!

**Pull request merged**